### PR TITLE
3.3.x test tweaks

### PR DESCRIPTION
**Pull request by Fenikkusu**
Hello!

* Type: bug fix | new feature | code quality

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:
A couple items going on here (One thing lead to another). First, I've renamed each codeception yaml to .dist.yaml and added the existing ones to the .gitignore. This allows us to override the default yamls without having to worry about not committing them (IE: I can add populate: true for testing the db module). Second, I corrected a misnamed drop tables. Third, I added a docker-compose.yml file that can be used with local testing. Finally (and the primary reason), I have made some slight changes to the ModelTestCase::setDb Method. The instance of change was due to the new testing I added and allows a child class of the passed name to work instead of just the exact class. Additionally, I added the ability to default the config to $this->getConfig() in the case $this->config has not yet been set. This makes it easier to add config settings. I also added a slight change to allow the setDb to default to the db config entry flat out if the adapter isn't set.

Thanks

### CODE REVIEW

File: tests/unit/Test/Traits/ModelTestCaseTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: fenikkusu
+ * Date: 2/11/18
+ * Time: 12:10 AM
+ */
```
Comment (sergeyklay) [inline tests/unit/Test/Traits/ModelTestCaseTest.php]: Could you remove this

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        image: schickling/beanstalkd
+        ports:
+            - "11300:11300"
+    memcached:
+        image: memcached
+        ports:
+            - "11211:11211"
+
+    mongodb:
+        image: "tutum/mongodb"
```
Comment (sergeyklay) [inline docker-compose.yml]: Could we use official image here?

**Review by sergeyklay — CHANGES_REQUESTED** [review]
We don't add changes directly to stable branches or `master`. Could you please send this PR to the `3.3.x` branch?

File: docker-compose.yml (hunk 2 above)
Comment (Fenikkusu) [reply docker-compose.yml]: Originally I was unable to figure out how to get the official image to create the correct DB. After some additionally digging I was able to find an undocumented method to use the official repo.

**Comment by Fenikkusu** [comment]
Updates Applied.

**Review by sergeyklay — CHANGES_REQUESTED** [review]
@Fenikkusu Looks good to me. Could you please amend README.md (https://github.com/phalcon/incubator/blob/master/tests/README.md) to add instruction about ignored Codeception's suites.

File: tests/_data/dump.sql [hunk 3]
```diff
@@ -1283,7 +1283,7 @@ CREATE TABLE `audit_detail` (
   PRIMARY KEY (id)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
-DROP TABLE if EXISTS `session`;
```
Comment (sergeyklay) [inline tests/_data/dump.sql]: 👍

**Comment by Fenikkusu** [comment]
I've updated the README.md file with both details about docker-compose and the advanced configuration. I wasn't sure if you wanted me to directly mention the ignored files or not. I personally didn't see a reason to directly mention the ignored files and instead opted for a 'you can also do this' type approach.

**Review by sergeyklay — APPROVED** [review]
Looks good to me

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**